### PR TITLE
TLS Revocation config

### LINKED ISSUE #8321: Helidon needs to support certificate revocation list (CRL) to reject client connections from clients whose certificate has been revoked
## Environment Details
* Helidon Version: 2.6.4
* Helidon SE or Helidon MP: SE
* JDK version: 17.0.4" 2022-07-19 LTS
* OS: Linux (OEL7)
* Docker version (if applicable): n/a

----------

## Problem Description

When the `tls.client-auth` is set to "REQUIRE" the client has to provide a certificate to connect to the server. The client's certificate could have been revoked (say because it has been compromised). We need a method to verify the certificate provided against a certificate revocation list (CRL) to ensure the certificate hasn't been revoked.

If the microservice has been configured with MTLS (`tls.client-auth` set to "REQUIRE") and a client presents a valid but revoked. 
Certificate, the connection will be accepted by Helidon currently. Instead, it should be rejected by Helidon based on a CRL that can be specified as part of the microservice configuration.
Note that the CRL should be dynamically loadable since CRL's keep getting updated periodically.

## Steps to reproduce
1. Create an MTLS microservice. 
2. Ensure that client connections are working properly
3. Revoke the client certificate
4. Connect with the revoked certificate. The connection will work currently. If Helidon can utilize a CRL then the connection shouldn't work as long as the CRL with the revoked certificate serial number listed in it, is supplied to the server.

### PR DESCRIPTION
**Pull request by Verdent**
### Description

Fixes #8321 

### Documentation

Generated documentation added

### CODE REVIEW

**Review by romain-grecourt — COMMENTED** [review]
Adding generated config documentation is not enough.
Can you provide a simplified usage example ?

File: common/tls/src/main/java/io/helidon/common/tls/ConfiguredTlsManager.java [hunk 1]
```diff
@@ -291,6 +299,37 @@ protected TrustManagerFactory createTmf(TlsConfig tlsConfig) {
         }
     }
 
+    protected void initializeTmf(TrustManagerFactory tmf, KeyStore keyStore, TlsConfig tlsConfig)
```
Comment (tomas-langer) [inline common/tls/src/main/java/io/helidon/common/tls/ConfiguredTlsManager.java:302]: `protected` should have javadoc.

Comment (tomas-langer) [reply common/tls/src/main/java/io/helidon/common/tls/ConfiguredTlsManager.java:302]: This method should not throw checked exceptions (to be consistent with other methods in this class)

File: common/tls/src/main/java/io/helidon/common/tls/RevocationConfigBlueprint.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.net.URI;
+import java.util.Optional;
+
+import io.helidon.builder.api.Option;
+import io.helidon.builder.api.Prototype;
+
+@Prototype.Blueprint
+@Prototype.Configured
+interface RevocationConfigBlueprint {
```
Comment (tomas-langer) [inline common/tls/src/main/java/io/helidon/common/tls/RevocationConfigBlueprint.java:27]: Missing class javadoc.

File: common/tls/src/main/java/io/helidon/common/tls/RevocationConfigBlueprint.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    @Option.Configured
+    @Option.DefaultBoolean(false)
+    boolean enabled();
+
+    /**
+     * Prefer CRL over OCSP.
+     * Default value is {@code false}.
+     *
+     * @return whether to prefer CRL over OCSP
```
Comment (tomas-langer) [inline common/tls/src/main/java/io/helidon/common/tls/RevocationConfigBlueprint.java:42]: would be nice to have a bit of explanation of what `CRL` and `OCSP` is (ie. CRL is a link to list of revoked certificates, OCSP is an endpoint to validate specific certificate)

File: common/tls/src/main/java/io/helidon/common/tls/RevocationConfigBlueprint.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     * Default value is {@code false}.
+     *
+     * @return whether to check only end-entity certificates
+     */
+    @Option.Configured
+    @Option.DefaultBoolean(false)
+    boolean checkOnlyEndEntity();
+
+    /**
+     * Enable fallback to the less preferred checking option.
```
Comment (tomas-langer) [inline common/tls/src/main/java/io/helidon/common/tls/RevocationConfigBlueprint.java:59]: Is this related to CRL and OCSP? If so, it should be mentioned

File: integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/DefaultOciCertificatesTlsManager.java [hunk 5]
```diff
@@ -198,6 +200,8 @@ boolean loadContext(boolean initialLoad) {
             return true;
         } catch (KeyStoreException e) {
             throw new IllegalStateException("Error while loading context from OCI", e);
+        } catch (InvalidAlgorithmParameterException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```
Comment (tomas-langer) [inline integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/DefaultOciCertificatesTlsManager.java:204]: RuntimeException should not be used, but I see it is used in other methods...

File: tests/integration/tls-revocation-config/src/test/resources/logging.properties [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+handlers=io.helidon.logging.jul.HelidonConsoleHandler
+java.util.logging.SimpleFormatter.format=%1$tY.%1$tm.%1$td %1$tH:%1$tM:%1$tS %4$s %3$s !thread!: %5$s%6$s%n
+
+.level=INFO
+
+sun.security.provider.certpath.level=FINEST
```
Comment (tomas-langer) [inline tests/integration/tls-revocation-config/src/test/resources/logging.properties:22]: this (FINEST or FINER log levels) should not be enabled in repo, please either comment out or remove

**Review by tomas-langer — CHANGES_REQUESTED** [review]

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**